runtime/gc_glue_java/ArrayletObjectModel.cpp

```diff
@@ -110,3 +115,38 @@ GC_ArrayletObjectModel::getArrayletLayout(J9Class* clazz, UDATA dataSizeInBytes,
 	}
 	return layout;
 }
+
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+bool
+GC_ArrayletObjectModel::freeDoubleMapping(MM_EnvironmentBase *env, J9IndexableObject *objectPtr, struct J9PortVmemIdentifier *identifier)
```

Comment:
perhpas this makes more sense to reside in MM_IndexableObjectAllocationModel (where doubleMapArraylets() already is)? again, perhaps, think about it.

Reply:
or the other one should be here. bottom line, together in same class

Reply:
There was an issue when I tried to put `freeDoubleMapping` in `MM_IndexableObjectAllocationModel` as I could not find a clean way to call `freeDoubleMapping` from CopyForwardScheme.cpp, GlobalMarkingScheme.cpp, and PartialMarkingScheme.cpp. Maybe move `doubleMapArraylets` to ArrayletObjectModel.cpp?

Reply:
Ok, I see MM_IndexableObjectAllocationModel  is instantiated locally on stack during allocation, and does not exist any more at the end of GC when you need to unmap it.

It believe this one still could be a static method in MM_IndexableObjectAllocationModel (you don't need any of original constructor perameters  like class etc - it's all embedded into the object itself).
doubleMapArraylets  semantically belongs more  to AllocationModel IMO and it's good to leave it there.

Not sure after all - even as-is it's not too bad

Reply:
True, that would work, I'll check and see how that looks.